# capítulo 13
##ejercicio 1

In [81]:
options(warn=-1)
library(plm)
library(car)
library(stats)
library(stargazer)
library(stats)
library(dplyr)
library(lmtest)
library(sandwich)

In [84]:

load("data/fertil1.RData")
fertil1 <- data
rm('data','desc')
mod13_1 <- lm(data = fertil1, kids ~ educ + age + I(age^2) + black + east + 
              northcen + west + farm + othrural + town +
              smcity + y74 + y76 + y78 + y80 + y82 + y84 )
car::linearHypothesis(mod13_1, c('farm', 'othrural', 'town', 'smcity'))

Res.Df RSS      Df Sum of Sq F        Pr(>F)  
1 1115   2697.104 NA       NA        NA       NA
2 1111   2685.898  4 11.20549  1.158765 0.327458

>i) El estadístico F de significación conjunta para las variables de condiciones del entorno a la edad de 16 años tiene un p-valor de 0.33 por lo cual no se puede rechazar la hipotesis de que todas valen cero.

In [ ]:

linearHypothesis(mod13_1, c('east', 'northcen', 'west'))


>ii) Las varibles de región del país a los 16 años tienen un p-valor para la prueba F de 0.029 por lo tanto si son signiticantes a un nivel del 97%. 

In [ ]:
u <- residuals(mod13_1)
test_model <- lm(I(u^2) ~  y74 + y76 + y78 + y80 + y82 + y84, data = fertil1 )
linearHypothesis(test_model, c('y74', 'y76', 'y78', 'y80', 'y82', 'y84'))


>iii) Existe suficiente evidencia estadística para aceptar heterocedasticidad en los errores. 

In [85]:
mod13_1 <- update(mod13_1, . ~ . + I(y74*educ) + I(y76*educ) + I(y78*educ)
                  + I(y80*educ) + I(y82*educ) + I(y84*educ))
summary(mod13_1, signif.stars = true, digits = 3)$coefficients
car::linearHypothesis(mod13_1, c("I(y74 * educ)", "I(y76 * educ)", "I(y78 * educ)", "I(y80 * educ)",
                                 "I(y82 * educ)", "I(y84 * educ)"))

Estimate     Std. Error  t value    Pr(>|t|)    
(Intercept)   -8.477302131 3.126360276 -2.7115564 6.800684e-03
educ          -0.022515231 0.053618261 -0.4199172 6.746276e-01
age            0.507465834 0.138921522  3.6528957 2.714583e-04
I(age^2)      -0.005524971 0.001570009 -3.5190710 4.506791e-04
black          1.074054877 0.173701015  6.1833541 8.817586e-10
east           0.206055930 0.133142996  1.5476288 1.219980e-01
northcen       0.348286721 0.121098738  2.8760557 4.104119e-03
west           0.177122131 0.167452303  1.0577468 2.904021e-01
farm          -0.072162198 0.147507965 -0.4892088 6.247910e-01
othrural      -0.191153859 0.175934037 -1.0865087 2.774910e-01
town           0.088229487 0.124535602  0.7084680 4.788041e-01
smcity         0.205357592 0.160210393  1.2817994 2.001819e-01
y74            0.946914919 0.904158933  1.0472881 2.951957e-01
y76            1.019963166 0.882033919  1.1563764 2.477772e-01
y78            1.805985151 0.951865866  1.8973106 5.804746e-02
y80            1.114183245 0.897601029  1.2412901 2.147620e-01
y82            1.199807328 0.876289067  1.3691913 1.712177e-01
y84            1.671260773 0.899050053  1.8589185 6.330439e-02
I(y74 * educ) -0.056424832 0.072560654 -0.7776230 4.369578e-01
I(y76 * educ) -0.092099696 0.070874767 -1.2994709 1.940534e-01
I(y78 * educ) -0.152387255 0.075281863 -2.0242227 4.318745e-02
I(y80 * educ) -0.097904853 0.070452286 -1.3896618 1.649115e-01
I(y82 * educ) -0.138944737 0.068371249 -2.0322100 4.237119e-02
I(y84 * educ) -0.176096976 0.069914918 -2.5187325 1.191811e-02

Res.Df RSS      Df Sum of Sq F        Pr(>F)   
1 1111   2685.898 NA       NA        NA        NA
2 1105   2664.435  6 21.46357  1.483569 0.1803366

>iv) Estos terminos representan la propensión del cambio de la educación en cada año. pero no son conjuntamente significativos. 

##Ejercicio 2

In [83]:
load("data/cps78_85.RData")
cps78_85 <- data
rm('data','desc')
mod13_2 <- lm(data = cps78_85, lwage ~ y85 + educ + y85educ + exper + expersq 
              +union + female + y85fem)
summary(mod13_2)$coefficients

Estimate      Std. Error   t value    Pr(>|t|)    
(Intercept)  0.4589328832 9.344850e-02  4.9110781 1.046281e-06
y85          0.1178062179 1.237817e-01  0.9517254 3.414502e-01
educ         0.0747209129 6.676431e-03 11.1917443 1.399530e-27
y85educ      0.0184605323 9.354169e-03  1.9735085 4.869344e-02
exper        0.0295843067 3.567312e-03  8.2931650 3.266999e-16
expersq     -0.0003994278 7.753912e-05 -5.1513072 3.075711e-07
union        0.2021318735 3.029449e-02  6.6722331 4.025626e-11
female      -0.3167086481 3.662145e-02 -8.6481733 1.876348e-17
y85fem       0.0850519706 5.130896e-02  1.6576435 9.768121e-02

>i)El coeficiente $y85$ en la ecuación $13.2$ refleja el control de el año 1985 en la
base de datos, a lo que representa la proporción de cambio del salario a través 
del tiempo correspondiente a 1985 cuando son hombres o en su caso la variable famale
tiene el valor 0 al igual que las demás variables binarias, en tanto a su relevancia
sería pequeña debido a que sólo explicaría una parte muy pequeña de la muestra.


In [86]:
mod13_22 <- lm(data=cps78_85, lwage~ y85 + educ + I(y85*I(educ-12)) + exper
               + expersq + union + female + y85fem)
summary(mod13_22)$coefficients

Estimate      Std. Error   t value   Pr(>|t|)    
(Intercept)            0.4589328832 9.344850e-02  4.911078 1.046281e-06
y85                    0.3393326055 3.400990e-02  9.977466 1.753665e-22
educ                   0.0747209129 6.676431e-03 11.191744 1.399530e-27
I(y85 * I(educ - 12))  0.0184605323 9.354169e-03  1.973508 4.869344e-02
exper                  0.0295843067 3.567312e-03  8.293165 3.266999e-16
expersq               -0.0003994278 7.753912e-05 -5.151307 3.075711e-07
union                  0.2021318735 3.029449e-02  6.672233 4.025626e-11
female                -0.3167086481 3.662145e-02 -8.648173 1.876348e-17
y85fem                 0.0850519706 5.130896e-02  1.657644 9.768121e-02

>ii) Queriendo saber cúal es el incremento porcentual en el salario de un hombre con
12 años de educación, entonces sería el cambio natural por el tiempo más el
cambio porcentual al tener 12 años de educación, como cambia el intercepto en 1985 
para un hombre con 12 años de educación, a lo el coeficiente estimado de y85 
con estas especificaciones es de .339 con una alta significatividad.
$0.3393±(1.96*0.03401)$
$$IC_{95} (0.2726404, 0.4059596)$$
El incremento nominal es de $33.93%$ y el intervalo de confianza al $95%$ es de $$IC_{95} (0.2726404, 0.4059596)$$ que es $27.26%$ a $40.60%$


>iii) 
ecuación de modelo nominal
$y85\; 0.118\;  se\;\simeq{.124}$
y la ecuación de modelo real
$$y85\; -0.383\; se\;\simeq{.124}$$
>aquí lo que muestra la disminución de los salarios reales a través de 1978 a 1985. a diferencia de la nominal que hay una aumento en los salarios, a lo que se ve que a subido pero no proporcional al aumento de los precios.



In [87]:
library(dplyr)
cps78_85 <- cps78_85 %>% mutate(wage=exp(lwage))
cps78_85 <- cps78_85 %>% mutate(rwage= if_else(year == 78, wage, wage/1.65))
mod13_2r <- lm(data=cps78_85, log(rwage)~y85 + educ + y85educ + exper
               + expersq + union + female + y85fem)
summary(mod13_2r)$coefficients


Estimate      Std. Error   t value   Pr(>|t|)    
(Intercept)  0.4589328832 9.344850e-02  4.911078 1.046281e-06
y85         -0.3829690700 1.237817e-01 -3.093906 2.026406e-03
educ         0.0747209129 6.676431e-03 11.191744 1.399530e-27
y85educ      0.0184605323 9.354169e-03  1.973508 4.869344e-02
exper        0.0295843067 3.567312e-03  8.293165 3.266999e-16
expersq     -0.0003994278 7.753912e-05 -5.151307 3.075711e-07
union        0.2021318735 3.029449e-02  6.672233 4.025626e-11
female      -0.3167086481 3.662145e-02 -8.648173 1.876348e-17
y85fem       0.0850519706 5.130896e-02  1.657644 9.768121e-02

>iv)
modelo 13.2
Residual standard error: $0.4127$ on $1075$ degrees of freedom
Multiple R-squared:  $0.4262$,	Adjusted R-squared:  $0.4219$

>modelo 13.2(iii)
Residual standard error: $0.4127$ on $1075$ degrees of freedom
Multiple R-squared:  $0.3562$,	Adjusted R-squared:  $0.3514$

>Comparando el $R^2$ del modelo con datos reales es de $.3562$ y para el nominal $.4262$, así se diferencian por a suma de los totales al cuadrado.


>v) Para 1978 había 168 afiliaciones de los 550 trabajadores representando un 30.5% afiliados, y para 1985 con una muestra de 534, solo habia 96 trabajadores afiliados, representando un 18%.
haciendo la comparación en porcentaje cayeron las afiliaciones por 12.5% en 7 años

In [88]:
mod13_2u <- lm(data = cps78_85, lwage ~ y85 + educ + y85educ + exper + expersq 
              +union + y85union + female + y85fem)
summary(mod13_2u)$coefficients

Estimate      Std. Error   t value      Pr(>|t|)    
(Intercept)  0.4588400102 9.458042e-02  4.851321416 1.407542e-06
y85          0.1179666851 1.262826e-01  0.934148535 3.504372e-01
educ         0.0747240557 6.697063e-03 11.157735441 1.978325e-27
y85educ      0.0184566351 9.377761e-03  1.968128179 4.931022e-02
exper        0.0295840011 3.569283e-03  8.288500628 3.392957e-16
expersq     -0.0003994210 7.758224e-05 -5.148356567 3.123788e-07
union        0.2022920534 3.908364e-02  5.175875546 2.706260e-07
y85union    -0.0003962278 6.104037e-02 -0.006491241 9.948220e-01
female      -0.3166908908 3.674048e-02 -8.619672833 2.371608e-17
y85fem       0.0850056955 5.182549e-02  1.640229478 1.012503e-01

>iv) $y85union$ al parecer tiene un disminución del alrededor de $.0004$ en la prima salarial, sin embargo al revisar su significatividad, notamos que su estadístico $t$ es muy bajo, a lo que no se rechaza $H0$, considerando que el coeficiente del término de interacción es 0

> vii) Se considera que no tiene algún inconveniente, como marca de manera explicativa, solo representa que hay una disminución en las personas que perciben este beneficio

##ejercicio 3


In [89]:
library('stats')
library('car')
load("data/kielmc.RData")
kielmc <- data
rm('data', 'desc')
mod3_1 <- lm(data = kielmc, lprice ~ y81 + ldist + y81ldist)
summary(mod3_1)$coefficients


Estimate    Std. Error t value     Pr(>|t|)    
(Intercept)  8.05846783 0.50843581 15.84952856 4.519980e-42
y81         -0.01131009 0.80506217 -0.01404872 9.888000e-01
ldist        0.31668901 0.05153226  6.14545179 2.391654e-09
y81ldist     0.04818618 0.08179290  0.58912422 5.561975e-01

>i)
>Para $\delta_1$ el signo es positivo es decir que si la vivienda está más lejana al 
incinerador, el precio de esta aumenta, y siendo $\beta_1$ positivo quiere decir que el
incinerador fue construido lejos de las casas más valiosas.

>ii)
$y81*\log(dist)$ representa el cambio en el valor de las casas cuando
el incinerado ya esta construido, dependiendo a la lejanía de la finca con el 
incinerador.

In [62]:
mod3_2 <- lm(data = kielmc, lprice ~ y81 + ldist + y81ldist + age + agesq + rooms + baths + lintst + lland + larea )
summary(mod3_2)$coefficients



ERROR: Error in is.data.frame(data): object 'kielmc' not found


>iii)
Al realizar las demás características de valuación de las casas, se muestran
que al parecer, las apreciaciones que están en función del incinerador, no son 
estadísticamente significativas, en realidad, el valor de las casas es más 
relevante por la cantidad de baños, el área y la antigüedad, que por la cercanía
o la existencia del incinerador.

>iv)
Es posible que entre las mismas variables exista una cercana tendencia 
que capte e interprete significatividad, sin embargo al meter las demás variables
que explican más a detalle el valor de las casas, no existe estadísticamente 
una relación del precio de las casas.


##Ejercicio 5

In [ ]:
library('stats')
library('car')
load("data/rental.RData")
rental <- data
rm('data','desc')
mod_rental <- lm(data=rental, lrent ~ y90 + lpop + lavginc + pctstu)
summary(mod_rental)$coefficients



>i) 
Para el coeficiente de y90, que es de .262 y es significativo, quiere decir
que manteniendo todo fijo, el efecto del tiempo en las rentas crecio en 10 años 
al rededor de un 26%, para pctstu siendo el porcentaje de estudiantes de 
la poblacion de la ciudad el durante el año escolar es positivo y ademas
significativo,a lo que hace setido que si incremente en 1% el porcentaje de 
estudiantes en la ciudad, las rentas aumentaran cerca de medio porciento.

>ii) 
Bajo el modelo
$$lrentit = y90t + lpopit + lavgincit + pctstuit + a_i + u_{it}$$ existe la parte de efectos
fijos que pueden o no estar en los errores lo que puede ocasionar autocorrelación 
lo que provocaría invalidades en los errores y las t_s no serían correctas.

In [ ]:
rental_pan <- pdata.frame(rental, index = c("city", "year"))

mod_rental_pan <- plm(data=rental_pan,lrent ~ y90 + lpop + lavginc 
                      + pctstu, model ='fd')
stargazer(mod_rental_pan, mod_rental, title="Resultados", align=TRUE, type = 'text')


>iii)modelo agregado $pctstu\; 0.005044$ , modelo con diferencia $pctstu \; 0.0187151$ apesar que pierde significatividad, el coeficiente aumenta alrededor de un 1.3%, al eliminar los efectos fijos, notamos una mayor fuerza en el aumento de del porcentaje de estudiantes por ciudad.

##Ejercicio 4

In [63]:
library('stats')
load("data/injury.RData")
injury <- data
rm('data','desc')
injuryky <- filter(injury,ky==1)
mod13_12ky <- lm(data=injuryky, ldurat~ afchnge + highearn + afhigh)  
mod13_12kyc <- lm(data=injuryky, ldurat~ afchnge + highearn + afhigh + male + married + head + neck + upextr + trunk + lowback + lowextr + occdis + manuf + construc)
stargazer(mod13_12ky, mod13_12kyc, title="Resultados", align=TRUE, type = 'text')



Resultados
                                   Dependent variable:                
                    --------------------------------------------------
                                          ldurat                      
                              (1)                       (2)           
----------------------------------------------------------------------
afchnge                      0.008                     0.011          
                            (0.045)                   (0.045)         
                                                                      
highearn                    0.256***                 0.176***         
                            (0.047)                   (0.052)         
                                                                      
afhigh                      0.191***                 0.231***         
                            (0.069)                   (0.070)         
                                                                 

>i) $afhigh\;0.191$ , $afhigh\;0.231$, agregando las variables de control hay un aumento de alrededor de 5%, el tiempo que reciben una compensación laboral, después del aumento en el límite de ganancias por semanas que cubría por compensación laborales para las personas con mayores ganancias, es decir que es propenso a que lo que más ganan sean cubiertos por mas semanas si sufren un accidente. 

>ii) reportando una $R^2$ de $0.0412$, $R^2 Adj:\;0.03868$. representando una explicación de $\log(Durat)$ de un 4.1%, a lo que se puede considerar que para explicar $\log(Durat)$ es más complicado y puede que haya terminamos que estemos dejando fuera del modelo.  

In [64]:
injurymi <- filter(injury,mi==1)
mod13_12mi <- lm(data=injurymi, ldurat~ afchnge + highearn + afhigh )
summary(mod13_12mi)$coefficients

Estimate   Std. Error t value   Pr(>|t|)     
(Intercept) 1.41273692 0.05671717 24.908454 3.974972e-115
afchnge     0.09738082 0.08478791  1.148522  2.509337e-01
highearn    0.16913878 0.10556762  1.602184  1.093227e-01
afhigh      0.19199063 0.15416988  1.245319  2.132066e-01

>iii) $afhigh\; 0.192$ hay un aumento en comparación a los datos de KY sin embargo no son significativos al 10%, puede ser además porque la muestra de kentucky es mucho mayor a la de Michigan.

##ejercicio 6

In [65]:
load("data/crime3.RData")
crime3 <- data

mod13_6 <- lm(data = crime3, lcrime ~ d78 + clrprc1 + clrprc2)
mod13_6aux <- lm(data = crime3, lcrime~ d78 + clrprc1 + I(clrprc1 + clrprc2))
mod13_6iii <- lm(data = crime3, clcrime ~ cavgclr)
stargazer(mod13_6, mod13_6aux, mod13_6iii, title="Resultados", align=TRUE, type = 'text')



Resultados
                                              Dependent variable:                          
                     ----------------------------------------------------------------------
                                         lcrime                             clcrime        
                               (1)                     (2)                    (3)          
-------------------------------------------------------------------------------------------
d78                          -0.055                  -0.055                                
                             (0.094)                 (0.094)                               
                                                                                           
clrprc1                     -0.018***                -0.001                                
                             (0.005)                 (0.010)                               
                                                                    

En el modelo auxiliar vimos que $\theta_1 = 0$ por lo tanto comprobamos que $\beta_1 = \beta_2$

##ejercico 7

In [66]:
library(dplyr)
library(car)
load("data/gpa3.RData")
gpa4 <- data
rm('data','desc')
mod1 <- lm(data=gpa4, trmgpa ~  spring + sat + hsperc + female + black + white + frstsem + tothrs + crsgpa + season)
summary(mod1)$coefficients

Estimate      Std. Error   t value    Pr(>|t|)    
(Intercept) -1.7528474371 0.3479048846 -5.0382950 5.943364e-07
spring      -0.0580066228 0.0480368104 -1.2075453 2.276182e-01
sat          0.0016984359 0.0001494168 11.3670984 1.174736e-27
hsperc      -0.0086610383 0.0010362790 -8.3578245 3.280021e-16
female       0.3504013324 0.0518524234  6.7576655 2.893544e-11
black       -0.2541494852 0.1229215861 -2.0675741 3.903632e-02
white       -0.0233146198 0.1173954218 -0.1985991 8.426324e-01
frstsem     -0.0346584832 0.0760344786 -0.4558259 6.486524e-01
tothrs      -0.0003389365 0.0007267233 -0.4663900 6.410772e-01
crsgpa       1.0478654858 0.1041144013 10.0645585 2.195313e-22
season      -0.0272903648 0.0490460388 -0.5564234 5.780941e-01

>i) $season\; -0.0273$, representa que cuando es temporada deportiva, el promedio de los estudiantes cae .027 puntos, no es estadísticamente significativo, 

>ii)si la habilidad no capturada está correlacionada con $season$, entonces estará sesgado y será inconsistente
  

In [67]:
library(plm)

gpa4.pan <- pdata.frame(gpa4, index = c("id","term"))
mod1df <- plm(trmgpa ~  spring + sat + hsperc + female + black + white + frstsem 
                + tothrs + crsgpa + season, data = gpa4.pan )
stargazer(mod1df, title="Resultados", align=TRUE, type = 'text')


Resultados
                 Dependent variable:    
             ---------------------------
                       trmgpa           
----------------------------------------
spring                 -0.237           
                       (0.206)          
                                        
frstsem                 0.019           
                       (0.069)          
                                        
tothrs                  0.012           
                       (0.014)          
                                        
crsgpa                1.136***          
                       (0.119)          
                                        
season                 -0.065           
                       (0.043)          
                                        
----------------------------------------
Observations             732            
R2                      0.208           
Adjusted R2            -0.603           
F Statistic    19.016*** (df = 5; 361)  
Note

>iii)Desaparecen las variables sat, hisperc, female, black, white.
>Ahora si el deporte está en temporada  de ser en datos agregados season -0.0273 a diferenciados season -0.0645, sin embargo sigue siendo no significativo. 

>iv)Posiblemente, considerando que exista un proceso de elección de clases, el hecho de meter más o menos materias por semestres podría influir en el desempeño, tanto escolar, como deportivo y además en el promedio 

##ejercicio 8

In [68]:
load("data/vote2.RData")
vote2 <- data
rm('data')
mod1c <- lm(data= vote2, cvote~ clinexp + clchexp + cincshr)
summary(mod1c)$coefficients

Estimate   Std. Error t value    Pr(>|t|)    
(Intercept) -2.5559359 0.6309071  -4.0512078 8.078352e-05
clinexp     -1.2915259 1.3821353  -0.9344425 3.515481e-01
clchexp     -0.5985306 0.7113430  -0.8414093 4.014318e-01
cincshr      0.1558681 0.0636806   2.4476550 1.551008e-02

>i) el intercepto dado por -2.56 es altamente significativo y la parte total de gastos  de campaña para los titulares del cargo con un coeficiente de 
cincshr 0.156 a lo que es significativo al 1% 

In [69]:
linearHypothesis(mod1c, c("clinexp","clchexp"))

Res.Df RSS      Df Sum of Sq F        Pr(>F)   
1 155    9282.273 NA       NA        NA        NA
2 153    9102.292  2 179.9807  1.512643 0.2236031

>ii) No rechaza H0, es decir que conjuntamente no son significativas, al igual que de manera individual, con un p-value de 0.2236

In [70]:
mod2c <- lm(data= vote2, cvote~cincshr)
summary(mod2c)$coefficients

Estimate   Std. Error t value   Pr(>|t|)    
(Intercept) -2.6811182 0.62527726 -4.287887 3.159612e-05
cincshr      0.2175863 0.03209341  6.779782 2.378697e-10

>iii) Un aumento del 10 por ciento en el total del gasto de campaña significaría un aumento de 2.18% en los votos al candidato en cargo

In [71]:
vote2r <- filter(vote2,rptchall==1)
mod2r <- lm(data=vote2r, cvote~cincshr)
summary(mod2r)$coefficients

Estimate    Std. Error t value   Pr(>|t|)  
(Intercept) -2.24982237 0.99860825 -2.252958 0.03148237
cincshr      0.09239118 0.08472626  1.090467 0.28391138

>iv) En este caso cincshr 0.092 a lo que un aumento en el 10% en el gasto del candidato a cargo, representaría un .9% de votos a su favor, sin embargo al tener una desviación estándar de .0847, considerado alto en este caso, no muestra significatividad, además que la determinación del voto a un candidato es más complejo, y el tratar de conocer la propensión de voto por peso gastado puede que esté sesgado y otro puento es la escasez de datos, es una muestra muy pequeña al solo tratar de revisar el comportamiento contra opositores que repiten. 

##ejercicio 9

In [80]:
library(stats)
library(plm)
load("data/crime4.RData")
crime4 <- data
rm('data')

crime4.pan <- pdata.frame(crime4, index = c("county","year"))

mod1c <- plm(data=crime4.pan,lcrmrte ~ d82 + d83 + d84 + d85 + d86 + d87 + lprbarr 
             + lprbconv + lprbpris + lavgsen + lpolpc, model = "fd")

mod2c <- plm(data=crime4.pan,lcrmrte ~ d82 + d83 + d84 + d85 + d86 + d87 + lprbarr 
             + lprbconv + lprbpris + lavgsen + lpolpc + lwcon + lwtuc + lwtrd 
             + lwfir + lwser + lwmfg + lwfed + lwsta + lwloc, model = "fd")

stargazer(mod1c,mod2c, title="Resultados", align=TRUE, type = 'text')


Resultados
                            Dependent variable:               
             -------------------------------------------------
                                  lcrmrte                     
                       (1)                      (2)           
--------------------------------------------------------------
d82                   0.008                    0.020          
                     (0.017)                  (0.021)         
                                                              
d83                 -0.084***                 -0.071**        
                     (0.023)                  (0.033)         
                                                              
d84                 -0.125***                 -0.089**        
                     (0.029)                  (0.043)         
                                                              
d85                 -0.122***                  -0.069         
                     (0.033)               

>i) Han variado de manera mínima los coeficientes, sin embargo si ha habido cambio en la significatividad de los parametros como: 

>d83 y d84 perdieron signifiatividad hasta .5%
>d85 y d86 perdieron su significatividad. 

>y las variables de Criminalidad su variación es minima y no altera su significatividad

In [73]:
library(car)
linearHypothesis(mod2c, c("lwcon","lwtuc","lwtrd","lwfir","lwser","lwmfg",
                          "lwfed", "lwsta", "lwloc"))

Res.Df Df Chisq    Pr(>Chisq)
1 529    NA       NA        NA 
2 520     9 11.26062 0.2582668

>ii) Son conjuntamente no significativo, al no rechazar H0 siendo las variables no son distintas a 0

##ejercicio 10

In [74]:
library(stats)
load("data/jtrain.RData")
jtrain <- data
rm("data")

library(plm)

jtrain.pan <- pdata.frame(jtrain, index = c ("fcode","year"))
mod2c <- plm(data=jtrain.pan,hrsemp~d88+d89+grant+grant_1+lemploy, model = "fd" )

summary(mod2c)



Oneway (individual) effect First-Difference Model

Call:
plm(formula = hrsemp ~ d88 + d89 + grant + grant_1 + lemploy, 
    data = jtrain.pan, model = "fd")

Unbalanced Panel: n = 135, T = 1-3, N = 390
Observations used in estimation: 255

Residuals:
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
-62.400  -5.130   0.704   0.059   4.640 137.000 

Coefficients:
        Estimate Std. Error t-value Pr(>|t|)    
d88     -0.75828    1.91854 -0.3952   0.6930    
d89      4.03449    3.18208  1.2679   0.2060    
grant   32.35374    2.88001 11.2339   <2e-16 ***
grant_1  1.17407    5.17112  0.2270   0.8206    
lemploy  0.34908    4.70046  0.0743   0.9409    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Total Sum of Squares:    177350
Residual Sum of Squares: 93091
R-Squared:      0.47512
Adj. R-Squared: 0.46672
F-statistic: 56.5735 on 4 and 250 DF, p-value: < 2.22e-16

>i)fueron usadas 225 observaciones... 

>ii)Es siginificativo el valor de grant y trata de decir que al recibir la el subsidio lo que provoca es que estaria empeorando el funcionamiento de la operación y desperdiciarias 32.36 más que si no lo tuvieras. 

>iii) puede que haya sido su poca variabilidad al mmomento de hacer las diferencias, lo que provocaria que si variabilidad estadistica se muy alta y este sea impresiso

##Ejericio 11

In [75]:
load("data/mathpnl.RData")
math_pane <- pdata.frame(data, index = c('distid','year'))
model11ii <- plm(math4 ~ year + lrexpp + lenrol + lunch, data = math_pane, model = 'fd')
model11.3 <- plm(math4 ~ year + lrexpp + lag(lrexpp) + lenrol + lunch , data = math_pane, model = 'fd')
stargazer(model11ii,model11.3, title="Resultados", align=TRUE, type = 'text')



Resultados
                            Dependent variable:                
             --------------------------------------------------
                                   math4                       
                        (1)                      (2)           
---------------------------------------------------------------
year1993             5.955***                  7.115***        
                      (0.518)                  (0.674)         
                                                               
year1994             12.431***                13.454***        
                      (0.759)                  (0.849)         
                                                               
year1995             25.198***                25.823***        
                      (1.067)                  (1.091)         
                                                               
year1996             25.918***                25.954***        
                      (1.200

>ii) En el primer modelo, si el gasto por alumno se incrementa en 10% generara una disminución en la tasa de quienes aprueban matemáticas de 3.5/10 $\simeq{0.35}\%$ 

>iii) El coeficiente que se agregó para $lrexpp_{t-1}$ es sginificativo a cualquier nivel y es de 4.7 míentras que el gasto en el año actaul se vuelve insignificante. Se puede decir que un aumento del gasto real de 100% en el año anterior incrementa en $\simeq{4.7}$ puntos la tasa de aprobados.

In [76]:
model11.3HAC <- plm(math4 ~ year + lrexpp + lag(lrexpp) + lenrol + lunch , 
                 data = math_pane, model = 'fd', vcov = vcovNW)
stargazer(model11.3HAC,model11.3, title="Resultados", align=TRUE, type = 'text')


Resultados
                               Dependent variable:     
                           ----------------------------
                                      math4            
                                (1)            (2)     
-------------------------------------------------------
year1993                      7.115***      7.115***   
                              (0.674)        (0.674)   
                                                       
year1994                     13.454***      13.454***  
                              (0.849)        (0.849)   
                                                       
year1995                     25.823***      25.823***  
                              (1.091)        (1.091)   
                                                       
year1996                     25.954***      25.954***  
                              (1.199)        (1.199)   
                                                       
year1997                     23.296*

In [77]:
#Errores estándar robustos a heterocedasticidad 
summary(model11.3, vcov = vcovHC)$coefficients

Estimate    Std. Error t-value    Pr(>|t|)     
year1993     7.11473904 0.8866415   8.0243693  1.406362e-15
year1994    13.45400752 1.0378371  12.9635055  1.610756e-37
year1995    25.82282027 1.3384444  19.2931594  1.164999e-78
year1996    25.95449745 1.3191999  19.6744228  1.467130e-81
year1997    23.29557622 1.4118028  16.5005880  7.897643e-59
year1998    38.09868327 1.5418688  24.7094199 8.560980e-124
lrexpp      -2.34138801 4.6799973  -0.5002969  6.168995e-01
lag(lrexpp)  4.70480596 2.8144097   1.6716848  9.468161e-02
lenrol       1.10832011 1.4471623   0.7658575  4.438161e-01
lunch        0.02687666 0.1482078   0.1813444  8.561084e-01

In [78]:
#según la documentación de plm::vcovNW asi deberia funcionar para
#errores estándar a heterocedasticidad y autocorrelación
#pero da Error in if (names(u)[i] == names(v)[j]) {: missing value where TRUE/FALSE needed
coeftest(model11.3, vcov.=vcovNW)


ERROR: Error in if (names(u)[i] == names(v)[j]) {: missing value where TRUE/FALSE needed


In [ ]:
##Ejercicio 12
load('data/murder.RData')
murder_pane <- pdata.frame(data, index('id','year'))
model12i <- lm(mrdrte ~ d93 + exec + unem, data = murder_pane)
summary(model12i)$coefficients


> i) 

In [ ]:
mur90 <- filter(murder_pane, year==90)
mur93 <- filter(murder_pane, year==93)
mur_diff <- mur90 - mur93
model12Fd <- lm(mrdrte ~ d93 + exec + unem, data =mur_diff)
summary(model12Fd)$coefficients


>ii)

In [ ]:
e <- residuals(model12Fd)
BP <-lm(I(e^2) ~  exec + unem, data = mur_diff)
linearHypothesis(BP, c('exec','unem'))
fit <- model12Fd$fitted.values
white.test<- lm(I(e^2) ~ fit + I(fit^2))
linearHypothesis(white.test, c('fit','I(fit^2)'))

>iii)no existe suficiente evidencia estadística para decir que hay heterocedasticidad en el modelo de primera diferencia, los p-value de las pruebas de Breusch Pagan y white son de 0.55 y 0.56 respectivamente. 

In [ ]:
coeftest(model12Fd, vcov= vcovHC)

>iv) El t Robusto a HC es -2.62 como se esperaba es más grande que el anterior porque los intervalos robustos son más amplios, pero no hace falta hacer estimaciones robustas por que no se encontró heterocedasticidad.

>vi) no hace falta hacer estimaciones robustas por que no se encontró heterocedasticidad por tal razon se prefiere el estadístico usual.

##Ejercicio 13

In [ ]:
load("data/wagepan.RData")
wage.pan <- data
rm(data)

mod1 <- lm(data=wage.pan, lwage~d81+d82+d83+d84+d85+d86+d87+educ+I(d81*educ)
           +I(d82*educ)+I(d83*educ)+I(d84*educ)+I(d85*educ)+I(d86*educ)+I(d87*educ)
           +union)
summary(mod1)$coefficients


>i) las variables de tiempo principalmente, además que las otras variables cambian poco en el tiempo.

In [ ]:
wage.panpan <- pdata.frame(wage.pan,index = c("nr","year"))

mod1c <- plm(data=wage.panpan,lwage~d81+d82+d83+d84+d85+d86+d87+educ+I(d81*educ)
             +I(d82*educ)+I(d83*educ)+I(d84*educ)+I(d85*educ)+I(d86*educ)+I(d87*educ)
             +union, model ="fd" )
summary(mod1c)

>ii)al hacer el modelo de diferencias, se anula la variable educación a lo que no puedo hacer la hypotesis estadistica, 

##Ejercicio 16

In [ ]:
county<-county <- data.frame(read.csv("data/countymurders.csv"))
sum = c(
  'Mean murdrate' = round(mean(county$murdrate),3),
  'Sd murdrate'   = round(sd(county$murdrate),3),
  '%murdrate = 0' = paste(round(nrow(filter(county, murdrate == 0))/nrow(county),4)*100,'%'),
  'Mean execs'    = round(mean(county$execs),3),
  'Sd execs'      = round(sd(county$execs),3),
  '%execs = 0'    = paste(round(nrow(filter(county, execs == 0))/nrow(county),4)*100,'%'),
  'max exces'     = max(county$execs),     
  'nrow exces > 0'= nrow(filter(county, execs > 0))
       )
sum


>i & ii) 
La media de ejecuciones por pena de muerte es muy baja por que de los 2197 condados únicamente se aplicó esta condena en 205 y el máximo fue de 7. Si no se tuvieran en cuenta los condados que no la aplicaron, el promedio sería de 1.25.

>iii)
Para poder aplicar sección cruzada por MCO se debe asumir que el error idiosincrático no está correlaciondo con las variables explicativas.

In [ ]:
library(stringr)
i=0
teta = NULL
for(year in unique(county$year)){
  teta[i<- i+1]= paste("y",str_sub(year, start= -2), sep = "_")   
  county[teta[i]] <-
    ifelse(county$year == year, 1, 0)
}
years <- select(county, c(teta[2:16]))
model16_4 <- lm(county$murdrate ~. + county$execs +
                lag(county$execs) + county$percblack + county$percmale 
                + county$perc1019 + county$perc2029, data = years)
summary(model16_4)$coefficients

>iv) En esta regresión, se observa que manteniendo los demás factores constantes, por cada pena capital, se incrementan los asesinatos en 0.16. Lo que puede estar pasando aquí es que los otros factores que se encuentran en el error idiosincrático inciden más que el número de ejecuciones que se realicen. 

>v) Es raro que tantos parámetros sean tan significativos, seguramente el modelo tiene problemas de autocorrelación y esto hace que se subestime los errores estándar y a su vez se sobre estime la significatividad de los parámetros.

In [ ]:
county_pane <- pdata.frame(county, index = c('countyid','year'))
model13_fd <- plm(murdrate ~ year + execs + lag(execs) + percblack + percmale + 
    perc1019 + perc2029, data = county_pane, model = 'fd')
summary(model13_fd)$coefficients

>vi) los estimadores de $execs$ y $execs_{t-1}$ son $\simeq{.003}$ y $\simeq{.028}$ es evidente a simple vista la diferencia con el modelo de sección cruzada, incluso cambia el signo.

>vii) No existe suficiente evidencia estadística para decir que la pena de muerte influye significativamente en el número de homicidios tanto $execs$ y $execs_{t-1}$ tienen valores $T$ muy pequeños. 